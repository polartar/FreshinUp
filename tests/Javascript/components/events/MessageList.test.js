import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import Component from '~/components/events/MessageList.vue'

describe('MessageList', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('defaults', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Filters', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })
    test('formatName Fitler with name', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          messages: [ { uuid: 1, owner: { name: 'John Smith' } } ]
        }
      })
      expect(wrapper.html()).toContain('JS')
    })
    test('formatName Fitler with null', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          messages: [ { uuid: 1, owner: { name: null } } ]
        }
      })
      expect(wrapper.html()).not.toContain('JS')
    })
  })
})
