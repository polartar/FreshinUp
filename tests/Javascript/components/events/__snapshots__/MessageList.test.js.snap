// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList Snapshots defaults 1`] = `
<div>
  <hr
    class="v-divider theme--light"
  />
   
</div>
`;
